cmd/unused-exporter/exporter.go

```diff
@@ -155,29 +152,13 @@ func (e *exporter) pollProvider(p unused.Provider) {
 			}
 
 			diskInfoByNamespace := make(map[string]*namespaceInfo)
-			for _, d := range disks {
-				diskLabels := []any{
-					slog.String("name", d.Name()),
-					slog.Int("size_gb", d.SizeGB()),
-					slog.Time("created", d.CreatedAt()),
-				}
-
-				meta := d.Meta()
-				if e.verbose {
-					diskMetaLabels := make([]any, 0, len(meta))
-					for _, k := range meta.Keys() {
-						diskMetaLabels = append(diskMetaLabels, slog.String(k, meta[k]))
-					}
-					diskLabels = append(diskLabels, diskMetaLabels...)
-				}
+			var ms []metric // TODO we can optimize this creation here and allocate memory only once
```

Comment:
Would this work? This is assuming we have one metric per disk. 

```suggestion
			ms := make([]metric, len(disks))
```

Reply:
Actually, I don't even think we want any of these... I agree with @inkel that the bug he's describing would definitely show up, but also, `len(disks)` isn't going to be the correct size either.  We're still going to end up re-allocating memory at some point because `disks` actually has:
- [one metric entry per disk](https://github.com/grafana/unused/blob/f07a4b0e9adac3894fb4a5cc16ba7b211cb20e19/cmd/unused-exporter/exporter.go#L179) for last used
- [three metric entry per scrape](https://github.com/grafana/unused/blob/f07a4b0e9adac3894fb4a5cc16ba7b211cb20e19/cmd/unused-exporter/exporter.go#L182) for operational metrics
- [one metric per namespace](https://github.com/grafana/unused/blob/f07a4b0e9adac3894fb4a5cc16ba7b211cb20e19/cmd/unused-exporter/exporter.go#L187) for count of unused disks per namespace
- [one metric per namespace AND disk type](https://github.com/grafana/unused/blob/f07a4b0e9adac3894fb4a5cc16ba7b211cb20e19/cmd/unused-exporter/exporter.go#L187) for disk size per type per namespace metric

So if we want to pre-allocate memory just one time we need to compute the sum of:

- len(disks)
- 3
- number of scraped namespaces
- number of disk types in each namespace 

It's further complicated because each namespace can have one or two entries in the `diskInfoByNamespace` map.

My opinion? It's not worth it to try and pre-calculate this just to save the memory allocation when the list gets too large.  Performance here isn't a concern and I think it's easier and more readable to just use `append`...

Reply:
Makes total sense, and definitely not worth to pursue such an improvement if it's not an issue.